src/Alley/WP/filter-side-effects.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Filter side effect functions
+ *
+ * (c) Alley <info@alley.co>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @package wp-filter-side-effects
+ */
+
+namespace Alley\WP;
+
+/**
+ * Wrapper for `add_filter()` and `generate_filter_side_effect()`.
+ *
+ * @param string   $tag             Filter name.
+ * @param callable $function_to_add Callback.
+ * @param int      $priority        Priority.
+ * @param int      $accepted_args   Accepted arguments.
+ * @return true|void
+ */
+function add_filter_side_effect( $tag, $function_to_add, $priority = 10, $accepted_args = 1 ) {
+	return add_filter( $tag, generate_filter_side_effect( $function_to_add ), $priority, $accepted_args );
+}
+
+/**
+ * Returns a function that can be added to a filter to use the filter like an action.
+ *
+ * @param callable $callback Callback function.
+ * @return callable Function that returns the filtered value after calling the callable.
+ */
+function generate_filter_side_effect( $callback ): callable {
+	return function ( ...$args ) use ( $callback ) {
+		$callback( ...$args );
+		return $args[0];
```

Comment:
I'm curious, if the filtered argument is an object, and the side effect function manipulates that object, is the returned value the modified or unmodified object? I suspect it's the _modified_ object since objects are passed by reference. If that's the case, it's worth consideration. I don't think this needs to account for that case, but perhaps documenting it in the README, saying that if the filtered argument is an object any modifications will persist. Food for thought.

Reply:
Correct, it would be the modified object. I tried a couple ways of phrasing this case, but couldn't find one that I liked, in part because I think it's implied by "thus behave like `add_action()` callbacks", since the same is true for those. I'll keep thinking about it and will open a new PR if I come up with something (or feel free to open one if there was a description you had in mind).